### PR TITLE
ButtonGroup: Fix button and icon button styling when tooltips are used on them

### PR DESCRIPTION
**Pull request by broccolinisoup**
Closes https://github.com/primer/react/issues/4129

#### Changed

- With this PR, we are wrapping button group items with a div. The main reason is to simplify CSS selection. You can [view the previous attempt to write a CSS](https://github.com/primer/react/pull/4210/files#diff-bcddaa67246d7fe17275dc05e761a986bec1a1bba786553401e81455ffaa63ad) to fix styling issues both for tooltip v1 and v2 on button group. 
- Enhanced CSS selection for the ButtonGroup children elements to ensure buttons and links still look the same if there are tooltips on them

### Rollout strategy

- [x] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [x] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: script/generate-e2e-tests.js [hunk 1]
```diff
@@ -293,6 +293,38 @@ const components = new Map([
           id: 'components-buttongroup-features--icon-buttons',
           name: 'Icon Buttons',
         },
+        {
+          id: 'components-buttongroup-features--icon-buttons-with-tooltip',
```
Comment (mperrotti) [inline script/generate-e2e-tests.js:297]: Is this the new way we prefer to create VRTs? I've been manually editing the `.test.ts` files 😅

File: packages/react/src/ButtonGroup/ButtonGroup.tsx [hunk 2]
```diff
@@ -1,37 +1,52 @@
...
-const ButtonGroup = styled.div`
+export type ButtonGroupProps = React.PropsWithChildren<SxProp & ComponentProps<typeof StyledButtonGroup>>
+
+const StyledButtonGroup = styled.div<SxProp>`
   display: inline-flex;
   vertical-align: middle;
   isolation: isolate;
 
-  && > * {
+  && :is(button, a) {
```
Comment (mperrotti) [inline packages/react/src/ButtonGroup/ButtonGroup.tsx:17]: I don't actually think we need `:is` for this. I think we can just select `button` and `a` directly

```suggestion
  button,
  a {
```

File: packages/react/src/ButtonGroup/ButtonGroup.tsx [hunk 3]
```diff
@@ -1,37 +1,52 @@
...
-
     :focus,
     :active,
     :hover {
       z-index: 1;
     }
   }
 
-  ${sx};
+  && > :first-child :is(button, a) {
```
Comment (mperrotti) [inline packages/react/src/ButtonGroup/ButtonGroup.tsx:29]: Same here - I don't think we need `&&` or `:is`, we can just directly select `button` and `a` children of `:first-child`

```suggestion
  > :first-child button,
  > :first-child a {
```

File: packages/react/src/ButtonGroup/ButtonGroup.tsx [hunk 4]
```diff
@@ -1,37 +1,52 @@
...
+    border-top-left-radius: ${get('radii.2')};
+    border-bottom-left-radius: ${get('radii.2')};
+  }
+
+  && > :last-child :is(button, a) {
```
Comment (mperrotti) [inline packages/react/src/ButtonGroup/ButtonGroup.tsx:34]: One more :)

```suggestion
  > :last-child button,
  > :last-child a {
```

**Review by mperrotti — COMMENTED** [review]
Looks great! I just made some suggestions to simplify the styling a little.

Just confirm that my suggestions actually work and then I'll approve.

**Comment by broccolinisoup** [comment]
closing in favour of https://github.com/primer/react/pull/5123

**Pull request closed without merging**